src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -205,22 +213,45 @@ def _clear_ready_queue(self) -> int:
     def join(self, timeout: Optional[float] = None) -> None:
         deadline = time.time() + timeout if timeout else None
         self._bulk_commit()
+
         while self._ongoing_billing_outcomes:
-            if deadline and deadline > time.time():
+            now = time.time()
+            time_left = now - deadline if deadline else None
+
+            if deadline and time_left and time_left > 0:
                 items_left = self._messages_ready_since_last_commit + len(
                     self._ongoing_billing_outcomes
                 )
-                logger.warning(f"join timed out, items in the queue left: {items_left}")
+                logger.warning(f"join timed out, items left in the queue: {items_left}")
                 break
-            self._mark_commit_ready()
-            self._bulk_commit()
+
+            future, metrics_msg = self._ongoing_billing_outcomes[0]
+            try:
+                if future:
+                    future.result(time_left)
+            except Exception:
+                if not future:
+                    continue
```

Comment:
What if we check `if future` outside the try-catch-block? Then this additional check would not be necessary.